### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by Lornakaboro**
## In this activity I did the following : 

- Stored data in files.
- Run a program using the command line.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,145 +1,136 @@
...
+    date = gets.chomp
+    @rentals << Rental.new(@books[book_index], @peoples[person_index], date)
+  end
+
+  def list_person_rentals(person_id)
+    @peoples.each do |person|
+      if person.id == person_id
+        puts 'Rentals:-'
+        person.rentals.each { |rental| puts "Date #{rental.date}, Book #{rental.book.title} by #{rental.book.author}" }
+      end
```
Comment (GzimAsani) [inline app.rb]: - [ ] When I am clicking the number 6 the app throws an error, I know this issue is not part of this millstone but it is very important issue if app breaks, so please check into that. Btw you nailed with the saving data great job :+1:

File: data_management_system.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    file = if File.exist?(path)
+             File.open(path, 'a')
+           else
+             File.open(path, 'w')
+           end
+    peoples.each do |people|
+      student(people, path, file) if people.is_a?(Student)
+      teacher(people, path, file) if people.is_a?(Teacher)
+    end
+  end
```
Comment (GzimAsani) [inline data_management_system.rb]: - [ ] After I try to create a person and then close the app with 7 I get the following error, kindly check into that. Also the name on the json file saves differently from what I input 
image
image
- [ ] It saves always the same name as the first one 
image

File: data_management_system.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    file = if File.exist?(path)
+             File.open(path, 'a')
+           else
+             File.open(path, 'w')
+           end
+    books.each do |book|
+      booking(book, path, file) if book.is_a?(Book)
+    end
+  end
+
```
Comment (GzimAsani) [inline data_management_system.rb]: - [ ] Same error with books, but my new book didn't get saved at all on the json 
image

File: data_management_system.rb (hunk 2 above)
Comment (GzimAsani) [reply data_management_system.rb]: - [ ] Even if I just run the app and exit it with 7 automatically one person gets saved, check into this as well please

File: data_management_system.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      person: {
+        id: rental.person.id,
+        name: rental.person.name,
+        age: rental.person.age,
+        parent_permission: rental.person.instance_variable_get(:@parent_permission)
+      }
+    }
+    file.puts(data.to_json)
+    file.close
+  end
```
Comment (GzimAsani) [inline data_management_system.rb]: - [ ] I was able to create a rental and saw on console but after exiting the app it wasn't saved on the json, kindly look into that

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Lornakaboro            :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with, everything looks good on the console :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb [hunk 5]
```diff
@@ -1,27 +1,43 @@
...
+    puts 'Please choose an option:'
+    puts '1. List Books'
+    puts '2. List People'
+    puts '3. Add Person'
+    puts '4. Add Book'
+    puts '5. Add Rental'
+    puts '6. List Rentals'
+    puts '7. Exit'
+    print 'Enter your choice: '
   end
```
Comment (OyePriscilla) [inline main.rb:15]: - [ ] I'm so sorry about this, I couldn't access your app functionality due to the error message below 👇. I observed you deleted the data folder, due to this; `people.json` is longer available. **Note: The data folder was there as of last review.**

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Lornakaboro 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Preserve the data app and load preserved data on startup ❎ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Lornakaboro 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Preserve the data app and load preserved data on startup ❎ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by osamaashraf6 — APPROVED** [review]
Hi @Lornakaboro 😊, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
MinionsGIF (2)

## Highlights🎉🎉

- [x] No linter errors present💯
- [x] The PR has a proper title and a descriptive body 👍

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**